### PR TITLE
Ryanjung create edit delete announcements

### PR DESCRIPTION
**Pull request by rymaju**
## Issue

Closes #[Write Issue Number Here]

## Changes

Go to the relevant issue and **tick the items that have been addressed by this PR**. Ensure you cover every item.

Briefly list the changes made to the code:

- Added support for this.
- And removed redunant use of that.
- Also this was included for reasons.

## Screenshots

Provide screenshots of any new components, styling changes, or pages. 
image

image
image
image

## Verification Steps

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves.

### CODE REVIEW

File: src/components/AnnouncementForm.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+const { TextArea } = Input;
+
+interface AnnouncementFormProps {
+  onFinish: (data: AnnouncementFormData) => void;
+}
+
+export interface AnnouncementFormData {
+  title: string;
+  description: string;
+  imageSrc?: string | null;
```
Comment (onb6) [inline src/components/AnnouncementForm.tsx:19]: ❓ Do you need to include the " | null" if you have it as an optional field?

File: src/components/AnnouncementForm.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      <Helmet>
+        <title>Create Announcement</title>
+      </Helmet>
+      <FormInitialText>
+        <Title level={3}>Create Announcement</Title>
+      </FormInitialText>
+      <Form
+        name="basic"
+        layout="vertical"
+        onFinish={async (data) => onFinish(await mapFormFieldsToData(data))}
```
Comment (onb6) [inline src/components/AnnouncementForm.tsx:50]: 👍

File: src/containers/createAnnouncement/index.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  AsyncRequestLoading,
+  AsyncRequestNotStarted,
+} from '../../utils/asyncRequest';
+
+const CreateAnnouncement: React.FC = () => {
+  const [createAnnouncementRequest, setCreateAnnouncementRequest] = useState<
+    AsyncRequest<void, any>
+  >(AsyncRequestNotStarted());
+
+  const params = useParams<{ id?: string }>();
```
Comment (onb6) [inline src/containers/createAnnouncement/index.tsx:26]: oo didn't know you could make this an optional id, that's cool!

**Review by onb6 — APPROVED** [review]
This is awesome!

File: src/api/protectedApiClient.ts [hunk 4]
```diff
@@ -144,6 +150,23 @@ const getEventRegistrations: (eventId: number) => Promise<Registration[]> = (
   ).then((res) => res.data.registrations);
 };
 
+const createAnnouncement = (
+  request: AnnouncementFormData,
+  eventId?: number,
+): Promise<void> => {
+  return AppAxiosInstance.post(
+    ProtectedApiClientRoutes.ANNOUNCEMENTS +
+      (eventId !== undefined ? `/${eventId}` : ''),
```
Comment (anirudh723) [inline src/api/protectedApiClient.ts:159]: I think `/${eventId || ''}` would also work

File: src/components/AnnouncementCard.tsx [hunk 5]
```diff
@@ -39,6 +39,7 @@ interface AnnoucementCardProps extends Announcement {
 }
 
 export const AnnouncementCard: React.FC<AnnoucementCardProps> = ({
+  id,
```
Comment (anirudh723) [inline src/components/AnnouncementCard.tsx:42]: Side note: `AnnoucementCardProps` should be `AnnouncementCardProps`

File: src/containers/deleteAnnouncement/index.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+          name="description"
+          content="Create an announcement for Lucy's Love Bus Events."
+        />
+      </Helmet>
+      <ContentContainer>
+        {asyncRequestIsFailed(deleteAnnouncementRequest) && (
+          <Alert
+            message="Error"
+            description={
+              'Announcement creation failed: ' +
```
Comment (anirudh723) [inline src/containers/deleteAnnouncement/index.tsx:51]: deletion*

File: src/containers/deleteAnnouncement/index.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      setDeleteAnnouncementRequest(AsyncRequestCompleted(undefined));
+    } catch (err) {
+      setDeleteAnnouncementRequest(AsyncRequestFailed(err));
+    }
+  };
+
+  return (
+    <>
+      <Helmet>
+        <title>Create Announcement</title>
```
Comment (anirudh723) [inline src/containers/deleteAnnouncement/index.tsx:40]: Delete*

File: src/containers/deleteAnnouncement/index.tsx [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        <meta
+          name="description"
+          content="Create an announcement for Lucy's Love Bus Events."
```
Comment (anirudh723) [inline src/containers/deleteAnnouncement/index.tsx:43]: Delete* and any others on this page

**Review by anirudh723 — APPROVED** [review]
Awesome job! Super clean. Didn't have many suggestions, so feel free to merge when you think its ready

**Pull request merged**